### PR TITLE
RBE: Improve error handling

### PR DESCRIPTION
**Pull request by bduffany**
* Treat "command not found" errors as FAILED_PRECONDITION errors so that Bazel does not retry them.
* Ensure that we populate the `Status.Message` field in the COMPLETED operation response. Previously, Bazel would show `Unavailable: <blank>` due to this field being missing.

**BEFORE** (Bazel prints this error output after several retries)

```
ERROR: /tmp/bb_output/external/com_google_protobuf/BUILD:513:10: Compiling src/google/protobuf/compiler/main.cc failed: (Exit 34): Remote Execution Failure:
Unavailable: 

java.io.IOException: com.google.devtools.build.lib.remote.ExecutionStatusException: UNAVAILABLE: 
        at com.google.devtools.build.lib.remote.GrpcRemoteExecutor.executeRemotely(GrpcRemoteExecutor.java:226)
        [...] (rest of Java stack trace omitted)
```

**AFTER** (Bazel prints this error output immediately after first attempt)

```
ERROR: /tmp/bb_output/external/com_google_protobuf/BUILD:513:10: Compiling src/google/protobuf/compiler/main.cc failed: (Exit 34): Remote Execution Failure:
Failed Precondition: failed to start docker container: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "/usr/bin/gcc": stat /usr/bin/gcc: no such file or directory: unknown

java.io.IOException: com.google.devtools.build.lib.remote.ExecutionStatusException: FAILED_PRECONDITION: failed to start docker container: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "/usr/bin/gcc": stat /usr/bin/gcc: no such file or directory: unknown
        at com.google.devtools.build.lib.remote.GrpcRemoteExecutor.executeRemotely(GrpcRemoteExecutor.java:226)
        [...] (rest of Java stack trace omitted)
```

---

**Version bump**: Patch 

**Related issues**:
* https://github.com/buildbuddy-io/buildbuddy-internal/issues/947
* https://github.com/buildbuddy-io/buildbuddy-internal/issues/1045

### CODE REVIEW

File: enterprise/server/test/integration/remote_execution/docker_in_firecracker/docker_in_firecracker_test.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package docker_in_firecracker_test
```
Comment (tylerwilliams) [inline enterprise/server/test/integration/remote_execution/docker_in_firecracker/docker_in_firecracker_test.go]: this too

File: enterprise/server/test/integration/remote_execution/docker_in_firecracker/BUILD [hunk 2]
```diff
@@ -0,0 +1,6 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
```
Comment (tylerwilliams) [inline enterprise/server/test/integration/remote_execution/docker_in_firecracker/BUILD]: do you want to save this for another CL?

File: enterprise/server/remote_execution/operation/operation.go [hunk 3]
```diff
@@ -96,10 +94,12 @@ func PublishOperationDone(stream StreamLike, taskID string, adInstanceDigest *di
 	return finalErr
 }
 
-func ExecuteResponseWithResult(ar *repb.ActionResult, summary *espb.ExecutionSummary, code codes.Code) *repb.ExecuteResponse {
-	rsp := &repb.ExecuteResponse{
-		Status: &statuspb.Status{Code: int32(code)},
-	}
+func ExecuteResponseWithCachedResult(ar *repb.ActionResult) *repb.ExecuteResponse {
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/operation/operation.go:97]: nit: could you comment what this does and the one below it (ExecuteResponseWithResult) too?

**Review by tylerwilliams — APPROVED** [review]
LGTM (w/ nits)

File: enterprise/server/remote_execution/operation/operation.go (hunk 3 above)
Comment (bduffany) [reply enterprise/server/remote_execution/operation/operation.go:99]: Done

File: enterprise/server/test/integration/remote_execution/docker_in_firecracker/docker_in_firecracker_test.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/test/integration/remote_execution/docker_in_firecracker/docker_in_firecracker_test.go]: Accidentally committed this -- removed

**Review by tylerwilliams — COMMENTED** [review]
this is a really nice change btw -- great to return more / better info to the user :)

**Pull request merged**